Editor/EditorCore/ShapeEditor.cs

```diff
@@ -136,44 +148,67 @@ void CreateSelectedShape(bool forceCloseWindow = false)
             var res = m_ShapeBuilders[s_CurrentIndex].Build();
             EditorUtility.InitObject(res);
             ApplyPreviewTransform(res);
-            DestroyPreviewObject();
+            DestroyPreviewObject(true);
 
             Undo.RegisterCreatedObjectUndo(res.gameObject, "Create Shape");
 
             if (forceCloseWindow || s_CloseWindowAfterCreateShape)
                 Close();
         }
 
-        void DestroyPreviewObject()
+        void DestroyPreviewObject(bool immediate)
+        {
+            if (immediate)
+                DestroyPreviewObjectInternal();
+            else
+                EditorApplication.delayCall += DestroyPreviewObjectInternal;
+        }
+
+        void DestroyPreviewObjectInternal()
         {
             if (m_PreviewObject != null)
             {
+                if (Selection.Contains(m_PreviewObject.gameObject))
+                    MeshSelection.RemoveFromSelection(m_PreviewObject.gameObject);
+
                 if (m_PreviewObject.GetComponent<MeshFilter>().sharedMesh != null)
                     DestroyImmediate(m_PreviewObject.GetComponent<MeshFilter>().sharedMesh);
 
                 DestroyImmediate(m_PreviewObject);
+
+                // When entering play mode the editor tracker isn't rebuilt before the Inspector redraws, meaning the
+                // preview object is still assumed to be in the selection. Flush the selection changes by rebuilding
+                // active editor tracker fixes this.
+                ActiveEditorTracker.RebuildAllIfNecessary();
```

Comment:
This line fails with Unity 2018.3 -> 2019.2.
`error CS0117: 'ActiveEditorTracker' does not contain a definition for 'RebuildAllIfNecessary'`